x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_view/index.tsx

```diff
@@ -4,25 +4,28 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import React from 'react';
+import { EuiBadgeGroup, EuiButton, EuiFlexGroup } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { EuiFlexGroup, EuiBadgeGroup, EuiButton } from '@elastic/eui';
 import { Streams } from '@kbn/streams-schema';
+import React from 'react';
+import { useStreamDetail } from '../../hooks/use_stream_detail';
 import { useStreamsAppParams } from '../../hooks/use_streams_app_params';
+import { StatefulStreamsAppRouter, useStreamsAppRouter } from '../../hooks/use_streams_app_router';
+import { RedirectTo } from '../redirect_to';
+import { ClassicStreamBadge, LifecycleBadge } from '../stream_badges';
 import { StreamDetailDashboardsView } from '../stream_detail_dashboards_view';
 import { StreamDetailOverview } from '../stream_detail_overview';
-import { useStreamDetail } from '../../hooks/use_stream_detail';
-import { ClassicStreamBadge, LifecycleBadge } from '../stream_badges';
 import { StreamsAppPageTemplate } from '../streams_app_page_template';
-import { StatefulStreamsAppRouter, useStreamsAppRouter } from '../../hooks/use_streams_app_router';
-import { RedirectTo } from '../redirect_to';
+import { StreamDescription } from './description';
 
 const getStreamDetailTabs = ({
   definition,
   router,
+  refreshDefinition,
 }: {
   definition: Streams.ingest.all.GetResponse;
   router: StatefulStreamsAppRouter;
+  refreshDefinition: () => void;
```

Comment:
I know we don't do it consistently, but what about using `useStreamDetail` to get access to `refreshDefinition` instead of passing it through all the places?

Reply:
yes will change